implementations/rust/ockam/ockam_channel/src/secure_channel_encryptor.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{ChannelKeys, SecureChannelError, SecureChannelVault};
+use ockam_core::async_trait;
+use ockam_core::compat::{boxed::Box, vec::Vec};
+use ockam_core::{Any, Encodable, Result, Route, Routed, TransportMessage, Worker};
+use ockam_node::Context;
+use tracing::debug;
+
+pub(crate) struct SecureChannelEncryptor<V: SecureChannelVault> {
+    keys: ChannelKeys,
+    remote_route: Route,
+    vault: V,
+}
+
+impl<V: SecureChannelVault> SecureChannelEncryptor<V> {
+    pub(crate) fn new(keys: ChannelKeys, remote_route: Route, vault: V) -> Self {
+        Self {
+            keys,
+            remote_route,
+            vault,
+        }
+    }
+
+    pub(crate) fn convert_nonce_from_u64(nonce: u64) -> ([u8; 8], [u8; 12]) {
+        let mut n: [u8; 12] = [0; 12];
+        let b: [u8; 8] = nonce.to_be_bytes();
+
+        n[4..].copy_from_slice(&b);
+
+        (b, n)
+    }
```

Comment:
A short comment explaining what this function does might be useful for the future